src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts

```diff
@@ -92,16 +94,17 @@ export class TextSubmissionViewerComponent implements OnChanges {
      * @param currentPlagiarismSubmission The submission to load the plagiarism information for.
      */
     private loadProgrammingExercise(currentPlagiarismSubmission: PlagiarismSubmission<TextSubmissionElement>) {
-        this.isProgrammingExercise = true;
-
         const domain: DomainChange = [DomainType.PARTICIPATION, { id: currentPlagiarismSubmission.submissionId }];
         this.repositoryService.getRepositoryContent(domain).subscribe({
             next: (files) => {
+                this.isProgrammingExercise = true;
                 this.loading = false;
                 this.files = this.programmingExerciseFilesWithMatches(files);
             },
             error: () => {
+                this.isProgrammingExercise = false;
                 this.loading = false;
+                this.fileContent = this.translateService.instant('artemisApp.plagiarism.cannotLoadFiles');
```

Comment:
Why did you decide to place this in the content section instead of using the alert system we have?
I feel like the viewer should have a default value that is shown whenever something is wrong or hasn't loaded yet. It should be independent of the situation, so always be the same. Examples:
- just empty
- Loading segment or a different icon
- different shade or background colour
- A combination of the above

The information about what happened, should be displayed as an alert.
What do you think?

Reply:
Thanks for reviewing this change. I think that warning of an error should be soft here as:
- single broken repo does not block the other comparisons to be viewed
- invalid repo state is not caused by the plagiarism detection logic 

In my opinion, this means that the user should not be stopped or annoyed by any form of a vibrant warning. When this message is displayed, the viewer has a different style and font, so that it is visible that this is not content of any file. Beside, this is a corner case scenario and I think adding more code to handle it is not worth it.

Reply:
Thanks for reviewing this change. I think that warning of an error should be soft here as:
- single broken repo does not block the other comparisons to be viewed
- invalid repo state is not caused by the plagiarism detection logic 

In my opinion, this means that the user should not be stopped or annoyed by any form of a vibrant warning. When this message is displayed, the viewer has a different style and font, so that it is visible that this is not content of any file. Besides, this is a corner case scenario and I think adding more code to handle it is not worth it.

WDYT?

Reply:
I think you misunderstood my point here:
When referring to "viewer" I didn't mean all of it. We could also name it "viewer segment".

The issue I have with the current implementation (at least if the screenshot is not out-of-date), is that the error message is inside the viewer segment where usually the code is placed and you actually have to look at it, read it and comprehend it to realise something went wrong.

Not showing anything and adding a call to the alert service is not more code but different one.
Adding a logic to have a different background or an icon is a little more code, but if you wanted to do it (which you obviously don't ^^) then it can be done with a flag, a class and a conditional for the icon.

Another option, I could agree with, would be adding an alert inside the box if you don't want to draw attention to the side.
https://ng-bootstrap.github.io/#/components/alert/examples
Just add a flag `loadingError: boolean` and show the error on condition `true`.
In that case, the error becomes clearly distinguishable from any other text. You don't necessarily have to choose a danger alert or similar.

It would be a rather custom solution, though, so not sure if you should check with the maintainers first before doing this.

So while I can totally get behind the idea of just doing a small thing for a small edge case, I hope you understand why I think it's visually not a good idea. I don't want you to develop a whole new application, just to close the gap between the current idea and the rest of the application design.